FRUI/FRUI/ViewControllers/AuthStepViewController.swift

```diff
@@ -348,6 +356,20 @@ class AuthStepViewController: UIViewController {
         self.view.endEditing(true)
         self.dismiss(animated: true, completion: nil)
     }
+    
+    @IBAction func closeButtonClicked(sender: UIButton) {
+        if let completion = self.atCompletion {
```

Comment:
Why the 3 separate conditions if they all result in the same function call?

Reply:
This was designed way back when I started the project and had very little knowledge of handling the generic types in Swift. This is something in my to-do list for refactoring a little bit in UI codebase. Essentially the reason for this conditions were to handle three different callback types (returning Token, AccessToken, and FRUser objects) in a single unified page.